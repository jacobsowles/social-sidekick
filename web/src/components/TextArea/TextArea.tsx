import classNames from 'classnames';
import React, { Component } from 'react';

import './TextArea.scss';

export interface TextAreaProps {
  className?: string;
  maxLength?: number;
  onChange: (event: any) => void;
  placeholder: string;
  value: string;
}

export interface TextAreaState {
  value: string;
}

class TextArea extends Component<TextAreaProps, TextAreaState> {
  constructor(props: TextAreaProps) {
    super(props);

    this.state = {
      value: props.value
    };
  }

  public render() {
    const { className, maxLength, placeholder } = this.props;

    return (
      <textarea
        aria-label={placeholder}
        className={classNames('text-area', className)}
        maxLength={maxLength}
        onChange={this.onChange}
        placeholder={placeholder}
        value={this.state.value}
      />
    );
  }

  private onChange = (event: any) => {
    event.preventDefault();
    this.setState({ value: event.target.value });
    this.props.onChange(event);
  };
}

export default TextArea;
