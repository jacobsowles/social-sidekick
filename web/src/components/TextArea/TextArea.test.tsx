import React from 'react';

import { init, shallow, toJson } from '@tests/component-test-base';
import TextArea, { TextAreaProps } from './TextArea';

describe('TextArea', () => {
  let props: TextAreaProps;
  init(() => <TextArea {...props} />);

  beforeEach(() => {
    props = {
      className: undefined,
      maxLength: undefined,
      onChange: jest.fn(),
      placeholder: undefined,
      value: undefined
    };
  });

  it('should render the component without crashing', () => {
    toJson(shallow()).should.matchSnapshot();
  });
});
