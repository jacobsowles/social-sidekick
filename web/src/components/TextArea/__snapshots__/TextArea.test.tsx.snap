// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea should render the component without crashing 1`] = `
<textarea
  className="text-area"
  onChange={[MockFunction]}
/>
`;
